posydon/binary_evol/SN/step_SN.py

```diff
@@ -2373,3 +2367,30 @@ def __call__(self, star, conserve_hydrogen_envelope=False):
             raise Exception("Need a NS or BH to apply `Sukhbold16_corecollapse`.")
 
         return float(m_rem), f_fb, state
+
+
+
+def check_SN_CO_match(star):
+    '''Check if the SN type matches the stellar state of the given star.
+    
+    Parameters
+    ----------
+    star : SingleStar object
+        Star object containing the star properties.
+        
+    Returns
+    -------
+    correct_SN_type : bool
+        True if the SN type matches the stellar state of the star.
+    '''
+    correct_SN_type = True
+    if star.state == 'WD' and star.SN_type != "WD":
+        correct_SN_type = False
+    elif (star.state == "NS" or star.state == "BH") and \
+            (star.SN_type != 'ECSN' and
+            star.SN_type != "CCSN" and
+            star.SN_type != 'PPISN'):
```

Comment:
Should we split this? E.g. a `BH` shouldn't be the outcome of an `ECSN` and I'd question, whether a `PPISN` can lead to a `NS`.